### PR TITLE
[doc] Update Dict Train Loader doc. 

### LINKED ISSUE #6578: Improve docs for multiple dataloaders
## 📚 Improve documentation

It would be great if you could extend [this page](https://pytorch-lightning.readthedocs.io/en/stable/advanced/multiple_loaders.html) by also explaining how to define the `training_step` and `validation_step` when using multiple dataloaders.

Currently, it's not that clear to me how to modify this. For example, when the `train_dataloader` is defined as follows:

```
class LitModel(LightningModule):

    def train_dataloader(self):

        loader_a = torch.utils.data.DataLoader(range(8), batch_size=4)
        loader_b = torch.utils.data.DataLoader(range(16), batch_size=4)
        loader_c = torch.utils.data.DataLoader(range(32), batch_size=4)
        loader_c = torch.utils.data.DataLoader(range(64), batch_size=4)

        # pass loaders as a nested dict. This will create batches like this:
        # {'loader_a_b': {'a': batch from loader a, 'b': batch from loader b},
        #  'loader_c_d': {'c': batch from loader c, 'd': batch from loader d}}
        loaders = {'loaders_a_b': {'a': loader_a, 'b': loader_b},
                   'loaders_c_d': {'c': loader_c, 'd': loader_d}}
        return loaders
```

, does the API of `training_step` still look like `training_step(self, batch, batch_idx)`? Or how can I access the multiple batches?

Thanks!

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes #6578

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/advanced/multiple_loaders.rst [hunk 1]
```diff
@@ -98,16 +98,14 @@ be returned
...
-            loader_c = torch.utils.data.DataLoader(range(32), batch_size=4)
-            loader_c = torch.utils.data.DataLoader(range(64), batch_size=4)
-
-            # pass loaders as a nested dict. This will create batches like this:
-            # {'loader_a_b': {'a': batch from loader a, 'b': batch from loader b},
-            #  'loader_c_d': {'c': batch from loader c, 'd': batch from loader d}}
-            loaders = {'loaders_a_b': {'a': loader_a, 'b': loader_b},
-                       'loaders_c_d': {'c': loader_c, 'd': loader_d}}
-            return loaders
+            loader_b = torch.utils.data.DataLoader(range(16), batch_size=2)
```
Comment (justusschock) [inline docs/source/advanced/multiple_loaders.rst:101]: Why to remove the option for nested dicts/lists from docs?

**Review by justusschock — APPROVED** [review]

File: docs/source/advanced/multiple_loaders.rst (hunk 1 above)
Comment (tchaton) [reply docs/source/advanced/multiple_loaders.rst:101]: Added below.

**Review by awaelchli — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]
